## Original (pre-optimisation)

In [1]:
import os, sys, copy
from pathlib import Path
ROOT=Path.cwd()

In [2]:
%%time

ORIG  = ROOT / "original_files"

# parameter point (H0=67.4, omega_b=0.0224, omega_cdm=0.120 fixed)
POINT = {
    "lsum": -0.902196, "ldiff": -1.20594, "l2": 10.3442, "l3": 0.677975,
    "asum": 0.512237, "adiff": 0.342735, "log10_fesc10": -0.834616,
    "alpha_esc": -0.338567, "log10Mcrit": 9.19371,
    "n_s": 0.943396, "logA": 3.32225,
}

# Adding the old files to path
sys.path.insert(0, str(ORIG))

import mcmc_uvlf_9params_old as mcmcmod          # the pre-optimisation file

info = copy.deepcopy(mcmcmod.info)
for k in ("output", "resume", "sampler"):
    info.pop(k, None)

from cobaya.model import get_model
model = get_model(info)
res = model.logposterior(POINT, return_derived=True)
dn = list(model.parameterization.derived_params().keys())
der = res.derived if isinstance(res.derived, dict) else dict(zip(dn, res.derived))
chi = {n: -2.0 * float(v) for n, v in zip(model.likelihood.keys(), res.loglikes)}

print("log-prior      = " + str(res.logprior))
print("log-like       = " + str(res.loglike))
print("log-posterior  = " + str(res.logpost))
print("chi2 per likelihood:")
for n in model.likelihood.keys():
    print("  " + n + "  =  " + str(chi[n]))
print("tau_e = " + str(der["tau_e"]))
print("A_s   = " + str(der["A_s"]))

[classy_reio] `classy` module loaded successfully from /home/aayush/.local/lib/python3.14/site-packages
[planck_2018_highl_plik.ttteee_lite] `clipy` module loaded successfully from /home/aayush/testing_bench/cobaya_packages/code/planck/clipy/clipy
----
clipy_0.15
Checking likelihood '/home/aayush/testing_bench/original_files/cobaya_packages/data/planck_2018/baseline/plc_3.0/hi_l/plik_lite/plik_lite_v22_TTTEEE.clik' on test data. got -292.286 expected -292.286 (diff -3.86314e-09)
----
before crop : TT 30 -> 2508, TE 30 -> 1996, EE 30 -> 1996
after crop  : TT 30 -> 2508, TE 30 -> 1996, EE 30 -> 1996
log-prior      = -11.25692808878494
log-like       = -557.467881170064
log-posterior  = -568.7248092588488
chi2 per likelihood:
  uvlf  =  50.875249742367686
  planck_2018_lowl.TT  =  22.453855322547327
  planck_2018_lowl.EE  =  397.0037733063246
  planck_2018_highl_plik.TTTEEE_lite  =  644.6028839688882
tau_e = 0.05931429340871399
A_s   = 2.772265641507676e-09
CPU times: user 1min 12s, sys: 

## Restart the kernel before Part 2
In order to make sure the previous version has not affected the later version, it is recommended to restart the kernel. It will also help in getting a correct baseline for the wall time comparison.

In [2]:
get_ipython().kernel.do_shutdown(restart=True)

{'status': 'ok', 'restart': True}

## Optimised Version

In [6]:

%%time
sys.path.insert(0, str(ROOT))
# parameter point (11 free inputs; H0=67.4, omega_b=0.0224, omega_cdm=0.120 fixed in the run)
POINT = {
    "lsum": -0.902196, "ldiff": -1.20594, "l2": 10.3442, "l3": 0.677975,
    "asum": 0.512237, "adiff": 0.342735, "log10_fesc10": -0.834616,
    "alpha_esc": -0.338567, "log10Mcrit": 9.19371,
    "n_s": 0.943396, "logA": 3.32225,
}


import mcmc_uvlf_9params as mcmcmod   # the copy of the repo-root file

info = copy.deepcopy(mcmcmod.info)
for k in ("output", "resume", "sampler"):
    info.pop(k, None)

from cobaya.model import get_model
model = get_model(info)
res = model.logposterior(POINT, return_derived=True)
dn = list(model.parameterization.derived_params().keys())
der = res.derived if isinstance(res.derived, dict) else dict(zip(dn, res.derived))
chi = {n: -2.0 * float(v) for n, v in zip(model.likelihood.keys(), res.loglikes)}

print("log-prior      = " + str(res.logprior))
print("log-like       = " + str(res.loglike))
print("log-posterior  = " + str(res.logpost))
print("chi2 per likelihood:")
for n in model.likelihood.keys():
    print("  " + n + "  =  " + str(chi[n]))
print("tau_e = " + str(der["tau_e"]))
print("A_s   = " + str(der["A_s"]))

[classy_reio] `classy` module loaded successfully from /home/aayush/.local/lib/python3.14/site-packages
[planck_2018_highl_plik.ttteee_lite] `clipy` module loaded successfully from /home/aayush/testing_bench/cobaya_packages/code/planck/clipy/clipy
----
clipy_0.15
Checking likelihood '/home/aayush/testing_bench/cobaya_packages/data/planck_2018/baseline/plc_3.0/hi_l/plik_lite/plik_lite_v22_TTTEEE.clik' on test data. got -292.286 expected -292.286 (diff -3.86314e-09)
----
before crop : TT 30 -> 2508, TE 30 -> 1996, EE 30 -> 1996
after crop  : TT 30 -> 2508, TE 30 -> 1996, EE 30 -> 1996
log-prior      = -11.25692808878494
log-like       = -557.4359746069173
log-posterior  = -568.6929026957022
chi2 per likelihood:
  uvlf  =  50.87665472938656
  planck_2018_lowl.TT  =  22.454365375308385
  planck_2018_lowl.EE  =  397.0037733063246
  planck_2018_highl_plik.TTTEEE_lite  =  644.537155802815
tau_e = 0.05930788559710075
A_s   = 2.772265641507676e-09
CPU times: user 29.5 s, sys: 1.32 s, total: 30.

Expected (measured): part 1 `tau_e ≈ 0.059314293`, part 2 `tau_e ≈ 0.059307886`; posterior differs by `≈ 0.03`, all per-likelihood chi2 are in margin of error.